graph2tac/loader/predict_server.py

```diff
@@ -319,15 +342,18 @@ def check_alignment(msg):
             unalignedDefinitions = unaligned_definitions,
         )
 
-def prediction_loop(predict_server, capnp_socket, record_file):
-    if predict_server.config.with_meter:
-        capnp_socket = tqdm.tqdm(capnp_socket)
+def prediction_loop(predict_server: PredictServer, capnp_socket: socket.socket, record_file: Optional[IO]):
+    
     message_generator = capnp_message_generator(capnp_socket, record_file)
+    if predict_server.config.with_meter:
+        message_generator_iterator = tqdm.tqdm(message_generator)
+    else:
+        message_generator_iterator = message_generator
     log_cnts = predict_server.log_cnts
     log_cnts.session_idx += 1
     log_cnts.thm_idx = -1
 
-    for msg in message_generator:
+    for msg in message_generator_iterator:
```

Comment:
That's a very strange name. Why both `generator` and `iterator`?

Reply:
Changed the name to just `message_iterator`.  By the way, here is what is going on.
```
message_generator = capnp_message_generator(capnp_socket, record_file)
```
returns a generator.  Generators have a method `send` which we use for checking alignment.  If we wrap the generator in `tqdm` then it no longer is a generator and we can't use `send`. So I need to keep around the inner generator as well as the wrapped iterator if we want to keep this `tqdm` functionality.

Reply:
Oh, I see

Reply:
I see. It seems kind of silly for `tqdm` not to support generators. But whatever.